### ISSUE TITLE
Cannot compile on windows 7 (64bit) for Python 3.5

### ISSUE THREAD

**Issue by laurensvdwiel** [pysam-developers/pysam#190]
I have tried with the setup.py as provided, the easy_install and pip approaches. All of these end in the same error.

Prior to receiving this error I have done a lot of digging on the web on where it may go wrong when compiling Python (3.5>=) modules on windows. The recommendations where to install a proper windows C++ compiler distribution and recommended was to install [Microsoft Visual Studio Community ENU 2015](https://www.visualstudio.com/en-us/post-download-vs?sku=community&clcid=0x409). The previous error was that [unable to find vcvarsall.bat](http://stackoverflow.com/questions/2817869/error-unable-to-find-vcvarsall-bat). Installing the Visual studio 2015 ENU fixed this.

This solved many of the problems I encountered that were related to the compiler not being able to be found.

The following error remains:

```
pip install pysam
Collecting pysam
  Using cached pysam-0.8.4.tar.gz
Installing collected packages: pysam
  Running setup.py install for pysam
    Complete output from command c:\python35\python.exe -c "import setuptools, tokenize;__file__='C:\\Users\\z950184\\AppData\\Local\\Temp\\pip-build-3zt_gg7m\\pysam\\setup.py';exec(compile(getattr(tokenize, 'open', open)(__file__).read().replace('\r\n', '\n'), __file__, 'exec'))" install --record C:\Users\z950184\AppData\Local\Temp\pip-f_3oc40u-record\install-record.txt --single-version-externally-managed --compile:
    running install
    running build
    running build_py
    creating build
    creating build\lib.win-amd64-3.5
    creating build\lib.win-amd64-3.5\pysam
    copying pysam\namedtuple.py -> build\lib.win-amd64-3.5\pysam
    copying pysam\Pileup.py -> build\lib.win-amd64-3.5\pysam
    copying pysam\version.py -> build\lib.win-amd64-3.5\pysam
    copying pysam\__init__.py -> build\lib.win-amd64-3.5\pysam
    creating build\lib.win-amd64-3.5\pysam\include
    copying pysam\include\__init__.py -> build\lib.win-amd64-3.5\pysam\include
    package init file 'htslib\__init__.py' not found (or not a regular file)
    package init file 'htslib\htslib\__init__.py' not found (or not a regular file)
    package init file 'samtools\__init__.py' not found (or not a regular file)
    package init file 'samtools\win32\__init__.py' not found (or not a regular file)
    copying pysam\calignedsegment.pxd -> build\lib.win-amd64-3.5\pysam
    copying pysam\calignmentfile.pxd -> build\lib.win-amd64-3.5\pysam
    copying pysam\cbcf.pxd -> build\lib.win-amd64-3.5\pysam
    copying pysam\cfaidx.pxd -> build\lib.win-amd64-3.5\pysam
    copying pysam\chtslib.pxd -> build\lib.win-amd64-3.5\pysam
    copying pysam\csamfile.pxd -> build\lib.win-amd64-3.5\pysam
    copying pysam\csamtools.pxd -> build\lib.win-amd64-3.5\pysam
    copying pysam\ctabix.pxd -> build\lib.win-amd64-3.5\pysam
    copying pysam\ctabixproxies.pxd -> build\lib.win-amd64-3.5\pysam
    copying pysam\cutils.pxd -> build\lib.win-amd64-3.5\pysam
    copying pysam\cvcf.pxd -> build\lib.win-amd64-3.5\pysam
    copying pysam\htslib_util.h -> build\lib.win-amd64-3.5\pysam
    copying pysam\pysam_stream.h -> build\lib.win-amd64-3.5\pysam
    copying pysam\pysam_util.h -> build\lib.win-amd64-3.5\pysam
    copying pysam\samfile_util.h -> build\lib.win-amd64-3.5\pysam
    copying pysam\tabix_util.h -> build\lib.win-amd64-3.5\pysam
    creating build\lib.win-amd64-3.5\pysam\include\htslib
    copying htslib\config.h -> build\lib.win-amd64-3.5\pysam\include\htslib
    copying htslib\hfile_internal.h -> build\lib.win-amd64-3.5\pysam\include\htslib
    copying htslib\version.h -> build\lib.win-amd64-3.5\pysam\include\htslib
    creating build\lib.win-amd64-3.5\pysam\include\htslib\htslib
    copying htslib\htslib\bgzf.h -> build\lib.win-amd64-3.5\pysam\include\htslib\htslib
    copying htslib\htslib\faidx.h -> build\lib.win-amd64-3.5\pysam\include\htslib\htslib
    copying htslib\htslib\hfile.h -> build\lib.win-amd64-3.5\pysam\include\htslib\htslib
    copying htslib\htslib\hts.h -> build\lib.win-amd64-3.5\pysam\include\htslib\htslib
    copying htslib\htslib\hts_defs.h -> build\lib.win-amd64-3.5\pysam\include\htslib\htslib
    copying htslib\htslib\kfunc.h -> build\lib.win-amd64-3.5\pysam\include\htslib\htslib
    copying htslib\htslib\khash.h -> build\lib.win-amd64-3.5\pysam\include\htslib\htslib
    copying htslib\htslib\khash_str2int.h -> build\lib.win-amd64-3.5\pysam\include\htslib\htslib
    copying htslib\htslib\klist.h -> build\lib.win-amd64-3.5\pysam\include\htslib\htslib
    copying htslib\htslib\knetfile.h -> build\lib.win-amd64-3.5\pysam\include\htslib\htslib
    copying htslib\htslib\kseq.h -> build\lib.win-amd64-3.5\pysam\include\htslib\htslib
    copying htslib\htslib\ksort.h -> build\lib.win-amd64-3.5\pysam\include\htslib\htslib
    copying htslib\htslib\kstring.h -> build\lib.win-amd64-3.5\pysam\include\htslib\htslib
    copying htslib\htslib\regidx.h -> build\lib.win-amd64-3.5\pysam\include\htslib\htslib
    copying htslib\htslib\sam.h -> build\lib.win-amd64-3.5\pysam\include\htslib\htslib
    copying htslib\htslib\synced_bcf_reader.h -> build\lib.win-amd64-3.5\pysam\include\htslib\htslib
    copying htslib\htslib\tbx.h -> build\lib.win-amd64-3.5\pysam\include\htslib\htslib
    copying htslib\htslib\vcf.h -> build\lib.win-amd64-3.5\pysam\include\htslib\htslib
    copying htslib\htslib\vcfutils.h -> build\lib.win-amd64-3.5\pysam\include\htslib\htslib
    copying htslib\htslib\vcf_sweep.h -> build\lib.win-amd64-3.5\pysam\include\htslib\htslib
    creating build\lib.win-amd64-3.5\pysam\include\samtools
    copying samtools\bam.h -> build\lib.win-amd64-3.5\pysam\include\samtools
    copying samtools\bam2bcf.h -> build\lib.win-amd64-3.5\pysam\include\samtools
    copying samtools\bam_endian.h -> build\lib.win-amd64-3.5\pysam\include\samtools
    copying samtools\bam_lpileup.h -> build\lib.win-amd64-3.5\pysam\include\samtools
    copying samtools\bam_plbuf.h -> build\lib.win-amd64-3.5\pysam\include\samtools
    copying samtools\bam_tview.h -> build\lib.win-amd64-3.5\pysam\include\samtools
    copying samtools\errmod.h -> build\lib.win-amd64-3.5\pysam\include\samtools
    copying samtools\kprobaln.h -> build\lib.win-amd64-3.5\pysam\include\samtools
    copying samtools\pysam.h -> build\lib.win-amd64-3.5\pysam\include\samtools
    copying samtools\sam.h -> build\lib.win-amd64-3.5\pysam\include\samtools
    copying samtools\sample.h -> build\lib.win-amd64-3.5\pysam\include\samtools
    copying samtools\samtools.h -> build\lib.win-amd64-3.5\pysam\include\samtools
    copying samtools\sam_header.h -> build\lib.win-amd64-3.5\pysam\include\samtools
    copying samtools\stats_isize.h -> build\lib.win-amd64-3.5\pysam\include\samtools
    copying samtools\version.h -> build\lib.win-amd64-3.5\pysam\include\samtools
    creating build\lib.win-amd64-3.5\pysam\include\samtools\win32
    copying samtools\win32\xcurses.h -> build\lib.win-amd64-3.5\pysam\include\samtools\win32
    copying samtools\win32\zconf.h -> build\lib.win-amd64-3.5\pysam\include\samtools\win32
    copying samtools\win32\zlib.h -> build\lib.win-amd64-3.5\pysam\include\samtools\win32
    Fixing build\lib.win-amd64-3.5\pysam\namedtuple.py build\lib.win-amd64-3.5\pysam\Pileup.py build\lib.win-amd64-3.5\pysam\version.py build\lib.win-amd64-3.5\pysam\__init__.py build\lib.win-amd64-3.5\pysam\include\__init__.py
    Skipping optional fixer: buffer
    Skipping optional fixer: idioms
    Skipping optional fixer: set_literal
    Skipping optional fixer: ws_comma
    Fixing build\lib.win-amd64-3.5\pysam\namedtuple.py build\lib.win-amd64-3.5\pysam\Pileup.py build\lib.win-amd64-3.5\pysam\version.py build\lib.win-amd64-3.5\pysam\__init__.py build\lib.win-amd64-3.5\pysam\include\__init__.py
    Skipping optional fixer: buffer
    Skipping optional fixer: idioms
    Skipping optional fixer: set_literal
    Skipping optional fixer: ws_comma
    running build_ext
    cythoning pysam/csamtools.pyx to pysam\csamtools.c
    building 'pysam.csamtools' extension
    creating build\temp.win-amd64-3.5
    creating build\temp.win-amd64-3.5\Release
    creating build\temp.win-amd64-3.5\Release\pysam
    creating build\temp.win-amd64-3.5\Release\samtools
    creating build\temp.win-amd64-3.5\Release\samtools\misc
    creating build\temp.win-amd64-3.5\Release\samtools\test
    creating build\temp.win-amd64-3.5\Release\win32
    creating build\temp.win-amd64-3.5\Release\htslib
    creating build\temp.win-amd64-3.5\Release\htslib\cram
    C:\Program Files (x86)\Microsoft Visual Studio 14.0\VC\BIN\amd64\cl.exe /c /nologo /Ox /W3 /GL /DNDEBUG /MD -D_FILE_OFFSET_BITS=64 -D_USE_KNETFILE= -Isamtools -Ipysam -Iwin32 -Ihtslib -Ic:\python35\include -Ic:\python35\include "-IC:\Program Files (x86)\Microsoft Visual Studio 14.0\VC\INCLUDE" "-IC:\Program Files (x86)\Microsoft Visual Studio 14.0\VC\ATLMFC\INCLUDE" "-IC:\Program Files (x86)\Windows Kits\10\include\10.0.10586.0\ucrt" "-IC:\Program Files (x86)\Windows Kits\NETFXSDK\4.6.1\include\um" "-IC:\Program Files (x86)\Windows Kits\10\include\10.0.10586.0\shared" "-IC:\Program Files (x86)\Windows Kits\10\include\10.0.10586.0\um" "-IC:\Program Files (x86)\Windows Kits\10\include\10.0.10586.0\winrt" /Tcpysam\csamtools.c /Fobuild\temp.win-amd64-3.5\Release\pysam\csamtools.obj -Wno-error=declaration-after-statement -DSAMTOOLS=1
    cl : Command line error D8021 : invalid numeric argument '/Wno-error=declaration-after-statement'
    error: command 'C:\\Program Files (x86)\\Microsoft Visual Studio 14.0\\VC\\BIN\\amd64\\cl.exe' failed with exit status 2

    ----------------------------------------
Command "c:\python35\python.exe -c "import setuptools, tokenize;__file__='C:\\Users\\z950184\\AppData\\Local\\Temp\\pip-build-3zt_gg7m\\pysam\\setup.py';exec(compile(getattr(tokenize, 'open', open)(__file__).read().replace('\r\n', '\n'), __file__, 'exec'))" install --record C:\Users\z950184\AppData\Local\Temp\pip-f_3oc40u-record\install-record.txt --single-version-externally-managed --compile" failed with error code 1 in C:\Users\z950184\AppData\Local\Temp\pip-build-3zt_gg7m\pysam
```

The error translates into [invalid numeric argument 'number', A number greater than 65,534 was specified as a numeric argument.](https://msdn.microsoft.com/en-us/library/kfk5bs0s.aspx)

Similar error encountered here: [cl D8021 error](http://stackoverflow.com/questions/13717551/1cl-command-line-error-d8021-invalid-numeric-argument-wno-deprecated-er). 

To ensure the problem was not in installing the requirements I manually installed (via pip) the cython module. This worked without errors, but did not fix the problem stated above.

**Comment by laurensvdwiel** [comment]
Similar problems encountered: 

https://groups.google.com/forum/#!topic/pysam-user-group/-eFI0qN1yyA
https://groups.google.com/forum/#!topic/pysam-user-group/tS7pkhA_4Pc

No solution as of yet.

**Comment by knyquist** [comment]
Same exact problem here, bookmarked in case solution is found.

**Comment by kyleabeauchamp** [comment]
@AndreasHeger  one of my previous projects has a working CI for windows using appveyor.  Not sure if that's useful to you or not.  

https://github.com/mdtraj/mdtraj/blob/master/devtools/appveyor-ci/run_with_env.cmd

**Comment by kyleabeauchamp** [comment]
Also this one: https://github.com/mdtraj/mdtraj/blob/master/appveyor.yml

**Comment by AndreasHeger** [comment]
Thanks! I will look into this after the pysam-0.8.5 release is out.

**Comment by AndreasHeger** [comment]
merged with #575
